src/object/obj_tx.c

```diff
@@ -1202,24 +1205,30 @@ dc_tx_classify_common(struct dc_tx *tx, struct daos_cpd_sub_req *dcsr,
 			if (dcu->dcu_ec_tgts == NULL)
 				D_GOTO(out, rc = -DER_NOMEM);
 
-			dcu->dcu_start_shard = start;
+			dcu->dcu_start_shard = grp_start;
```

Comment:
grp_start is start shard idx (offset in the obj shard array), not the start shard_id (OID's id_shard), in OSA they could be different.
but dcu->dcu_start_shard should use start shard_id right?
not a problem introduced in this patch, seems original code with problem for OSA case?
if above is correct please fix it in a following patch, thx

Reply:
The dcu_start_shard is used to calculate the parameter "tgt_idx" for obj_ec_tgt_oiod_get(). After apply Di's last patch (PR#9873), here, the dcu_start_shard should be changed as shard identifier instead of the shard index.

@wangdi1 , do you want to fix it in the patch? or you want me (or yourself) to fix in another patch?

Reply:
oh, I will fix it in the following patch. thanks.